### PR TITLE
Enable project configuration viewing in viewer mode

### PR DESCRIPTION
**Pull request by umarphaarook**
| Q                       | A                                |
| ----------------------- | -------------------------------- |
| Fixed issues?           | Related to #297 |
| Tests added + passed?   | No                              |
| Changeset added?        | Yes                              |
| Documentation PR link   |                                  |
| Any dependency changes? | No                               |

### CODE REVIEW

File: packages/legend-studio/src/stores/ViewerStore.tsx [hunk 1]
```diff
@@ -198,6 +199,16 @@ export class ViewerStore {
             undefined,
           )) as Entity[];
       }
+
+      // Get Project Config
+      const projectConfiguration: PlainObject<ProjectConfiguration> =
+        flowResult(
+          yield this.editorStore.applicationStore.networkClientManager.sdlcClient.getConfiguration(
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/stores/ViewerStore.tsx:206]: Did we account for viewer mode where workspaceId would not be defined like through a version or revisionID?

Comment (umarphaarook) [reply packages/legend-studio/src/stores/ViewerStore.tsx:206]: Nope, the logic within `ProjectConfigurationEditor` component always expects `projectConfiguration` to be present in the store. `EditorStore` does load the `projectConfiguration` in standard mode while `ViewerStore` doesn't, hence the crash on the app whenever user try to view project `config`  in viewer mode.

Comment (MauricioUyaguari) [reply packages/legend-studio/src/stores/ViewerStore.tsx:206]: yeah understood, my question was more when no workspace is provided, which occurs when you are in viewer mode for released version or on specific revision.

Comment (umarphaarook) [reply packages/legend-studio/src/stores/ViewerStore.tsx:206]: I couldn't find any logic that handles that, so I assumed we don't

Comment (MauricioUyaguari) [reply packages/legend-studio/src/stores/ViewerStore.tsx:206]: lets test your changes locally with a revision/version id and make sure this still works.

File: packages/legend-studio/src/stores/ViewerStore.tsx [hunk 2]
```diff
@@ -194,12 +197,35 @@ export class ViewerStore {
...
+            ),
+          ])) as [Entity[], PlainObject<ProjectConfiguration>];
+          entities = result[0];
+          projectConfiguration = result[1];
+          this.editorStore.projectConfigurationEditorState.setProjectConfiguration(
+            ProjectConfiguration.serialization.fromJson(projectConfiguration),
+          );
+          // make sure we set the original project configuration to a different object
+          this.editorStore.projectConfigurationEditorState.setOriginalProjectConfiguration(
+            ProjectConfiguration.serialization.fromJson(projectConfiguration),
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/stores/ViewerStore.tsx:219]: we should only be doing the project configuration once before setting them

File: packages/legend-studio/src/stores/ViewerStore.tsx [hunk 3]
```diff
@@ -152,7 +153,9 @@ export class ViewerStore {
           `Can't have both version ID and revision ID specified for viewer mode`,
         );
       }
+
       let entities: Entity[] = [];
+      let projectConfiguration: PlainObject<ProjectConfiguration>;
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/stores/ViewerStore.tsx:158]: can we name this `rawProjectConfiguration`
and the serialized one `projectConfiguration`
thats more inline with what we do in the rest of the app

File: packages/legend-studio/src/stores/ViewerStore.tsx [hunk 4]
```diff
@@ -194,12 +196,39 @@ export class ViewerStore {
...
+              undefined,
+            ),
+            this.editorStore.applicationStore.networkClientManager.sdlcClient.getConfiguration(
+              this.editorStore.sdlcState.currentProjectId,
+              undefined,
+            ),
+          ])) as [Entity[], PlainObject<ProjectConfiguration>];
+          entities = result[0];
+          const rawProjectConfiguration = result[1];
+          const serializedProjectConfiguration =
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/stores/ViewerStore.tsx:213]: `serializedProjectConfiguration` -> `projectConfiguration`

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**